ion/unmarshal.go

```diff
@@ -646,21 +690,27 @@ func (d *Decoder) decodeStructToMap(v reflect.Value) error {
 	}
 
 	for d.r.Next() {
-		name := d.r.FieldName()
-		if err := d.decodeTo(subv); err != nil {
+		fieldName, err := d.r.FieldName()
+		if err != nil {
 			return err
 		}
 
-		var kv reflect.Value
-		switch t.Key().Kind() {
-		case reflect.String:
-			kv = reflect.ValueOf(*name)
-		default:
-			panic(fmt.Sprintf("the key for map to hold field name must be of type string. Found: %v", t.Key().Kind().String()))
-		}
+		if fieldName != nil && fieldName.Text != nil {
+			if err := d.decodeTo(subv); err != nil {
+				return err
+			}
 
-		if kv.IsValid() {
-			v.SetMapIndex(kv, subv)
+			var kv reflect.Value
+			switch t.Key().Kind() {
+			case reflect.String:
+				kv = reflect.ValueOf(*fieldName.Text)
```

Comment:
spacing looks off

Reply:
The spacing appears normal in Goland, it is tab spacing